server/src/main/java/org/opensearch/index/shard/RemoteStoreRefreshListener.java

```diff
@@ -193,6 +202,8 @@ private synchronized void syncSegments(boolean isRetry) {
                                 OnSuccessfulSegmentsSync();
                                 final long lastRefreshedCheckpoint = ((InternalEngine) indexShard.getEngine()).lastRefreshedCheckpoint();
                                 indexShard.getEngine().translogManager().setMinSeqNoToKeep(lastRefreshedCheckpoint + 1);
+
+                                checkpointPublisher.publish(indexShard, checkpoint);
```

Comment:
Does publish throw any exception that we need to handle here?

Reply:
Correct any transport level exceptions would be handled in the publisher implementation.